api/core/workflow/nodes/loop/loop_node.py

```diff
@@ -534,10 +534,8 @@ def _get_segment_for_constant(var_type: SegmentType, original_value: Any) -> Seg
             else:
                 logger.warning("unexpected value for LoopNode, value_type=%s, value=%s", original_value, var_type)
                 value = []
-        elif var_type == SegmentType.ARRAY_BOOLEAN:
-            value = original_value
         else:
-            raise AssertionError("this statement should be unreachable.")
+            value = original_value
```

Comment:
I suggest keep the `else` branch raising `AssertionError("this statement should be unreachable.")`, and adding another branch to handle non-array types explicitly, like the following code:

```py
if not var_type.is_array_type():
    value = original_value
elif var_type in [SegmentType.ARRAY_NUMBER, SegmentType.ARRAY_OBJECT, SegmentType.ARRAY_STRING]:
    ...
elif elif var_type == SegmentType.ARRAY_BOOLEAN:
    ...
else:
    raise AssertionError("this statement should be unreachable.")
```